syndesis/plugins/org.fusesource.ide.syndesis.extension.core/src/org/fusesource/ide/syndesis/extensions/core/util/IgniteVersionMapper.java

```diff
@@ -11,24 +11,80 @@
 package org.fusesource.ide.syndesis.extensions.core.util;
 
 import java.util.HashMap;
+import java.util.Iterator;
 import java.util.Map;
+import java.util.Map.Entry;
 
+import org.apache.maven.model.Dependency;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.fusesource.ide.camel.model.service.core.util.OnlineArtifactVersionSearcher;
 import org.fusesource.ide.foundation.core.util.OnlineVersionMapper;
+import org.fusesource.ide.foundation.core.util.Strings;
 
-final class IgniteVersionMapper extends OnlineVersionMapper {
+public class IgniteVersionMapper extends OnlineVersionMapper {
 	
-	private static final String URL_IGNITE_VERSIONS_FILE = BASE_REPO_CONFIG_URI+"ignite.properties";
+	private static final String URL_IGNITE_VERSIONS_FILE = BASE_REPO_CONFIG_URI + "igniteVersionToDisplayName.properties";
+	private static final String SNAPSHOT_POSTFIX = "-SNAPSHOT";
 
-	IgniteVersionMapper() {
+	public IgniteVersionMapper() {
 		super("org.jboss.tools.fuse.ignite.url", IgniteVersionMapper.URL_IGNITE_VERSIONS_FILE);
 	}
 
 	@Override
 	protected Map<String, String> createFallbackMapping() {
 		Map<String, String> mapping = new HashMap<>();
-		mapping.put(SyndesisExtensionsUtil.KEY_CAMEL_VERSION, "2.20.1");
-		mapping.put(SyndesisExtensionsUtil.KEY_SPRING_BOOT_VERSION, "1.5.8.RELEASE");
-		mapping.put(SyndesisExtensionsUtil.KEY_SYNDESIS_VERSION, "1.2.3");
+		mapping.put("1.3-SNAPSHOT", "1.3-SNAPSHOT (Fuse Ignite TP4)");
 		return mapping;
 	}
+	
+	/* (non-Javadoc)
+	 * @see org.fusesource.ide.foundation.core.util.OnlineVersionMapper#consolidateMapping(java.util.Map)
+	 */
+	@Override
+	protected Map<String, String> consolidateMapping(Map<String, String> mapping) {
+		Map<String, String> map = super.consolidateMapping(mapping);
+		consolidateSnapshots(map, new NullProgressMonitor());		
+		return map;
+	}
+	
+	public static void consolidateSnapshots(Map<String, String> map, IProgressMonitor monitor) {
+		// check for -SNAPSHOT versions
+		Map<String, String> toAdd = new HashMap<>();
+		Iterator<Entry<String, String>> it = map.entrySet().iterator();
+		while (it.hasNext()) {
+			Entry<String, String> entry = it.next();
+			String version = entry.getKey();
+			String label = entry.getValue();
+			if (isSnapshot(version) && !SyndesisVersionUtil.isSyndesisVersionExisting(version, monitor)) {
```

Comment:
the monitor is not well subdivised in this method.

Maybe just raise a ticket to work on it after code freeze as not completely trivial to handle it correctly as there is while loop and some conditionals

Reply:
https://issues.jboss.org/browse/FUSETOOLS-2888